corehq/motech/repeaters/tests/test_couchsqlmigration.py

```diff
@@ -461,6 +461,16 @@ def test_migrate_record_with_no_attempts(self):
         obj = SQLRepeatRecord.objects.get(couch_id=doc_id)
         self.assertFalse(obj.attempts)
 
+    def test_migrate_record_erroneous_next_check(self):
+        doc, _ = self.create_repeat_record()
+        doc.update(succeeded=True, next_check=datetime.utcnow().isoformat() + 'Z')
```

Comment:
The outcome would be the same for a repeat record that succeeded or was cancelled. What is the expected behavior if a repeat record was in a different state with an erroneous next check value? Or do you not expect to encounter that situation? Should a hard failure be documented in tests as well?

Reply:
So far I have only encountered Couch documents that have `succeeded == True` and a non-null `next_check` value. If the state was _pending_ or _failed_ and `next_check` was null then that would also be a problem, but I have not encountered that (yet).

> Should a hard failure be documented in tests as well?

Not quite following. The database constraint on `next_check` will cause a hard failure if the record state is not compatible with the value of `next_check`. That is covered by various tests, although not tested explicitly. The check constraint is declarative and I don't think we need to add more tests around it

https://github.com/dimagi/commcare-hq/blob/17cfe64fc09b82e86d19870bfc3d6f8bb6cd8735/corehq/motech/repeaters/models.py#L1369-L1374

Reply:
Wasn't aware of the constraint. Thanks for the thorough response!